commentbanner.py

```diff
@@ -1,50 +1,69 @@
-
-"""This module provides the bannerComment plugin
-class and supporting methods."""
-
 import sublime
 import sublime_plugin
-# import textwrap
+
+
+class Prefs:
+
+    @staticmethod
+    def read():
+        settings = sublime.load_settings('CommentBanner.sublime-settings')
+        Prefs.character = settings.get('character', '*')
+        Prefs.banner_width = settings.get('banner_width', 80)
+
+    @staticmethod
+    def load():
+        settings = sublime.load_settings('CommentBanner.sublime-settings')
+        settings.add_on_change('character', Prefs.read)
+        settings.add_on_change('banner_width', Prefs.read)
+        Prefs.read()
 
 
 class BannerCommand(sublime_plugin.TextCommand):
-    """create a fancy comment box around the text selection.
-    - supports multi line selection
-    - supports multi cursor selection
+    """Create a comment box around the selection.
+
+    * supports multi-line selection
+    * supports multi-cursor selection
     """
-    ROW_LENGTH = 77  # 80 - commenttags (like //) and space
+
+    def __init__(self, *args, **kw):
+        super(BannerCommand, self).__init__(*args, **kw)
+        Prefs.load()
 
     def run(self, edit):
-        for region in self.view.sel():
-            bannerText = self.view.substr(region)
-            if bannerText:
-                print(region.begin())
-                self.view.erase(edit, region)
-                self.view.insert(edit, region.begin(),
-                                 self.full_screen_banner(bannerText))
-                region_len = (region.begin() +
-                              self.ROW_LENGTH*(2+len(self.lines)))
-                self.view \
-                    .selection \
-                    .add(sublime.Region(region.begin(), region_len))
-        # add the language dependend comment characters
-        self.view.run_command("toggle_comment", False)
-
-        # remove the selection of the cursor
-        self.view.run_command("move", {"by": "characters", "forward": True})
-
-    def full_screen_banner(self, string, symbol='*'):
-        def outer_row():
-            return (self.ROW_LENGTH - 1) * symbol + '\n'
-
-        def inner_row():
-            result = ""
-            self.lines = string.splitlines()
-            # textwrap.wrap(string)
+        view = self.view
+
+        for region in view.sel():
+            lines = view.full_line(region)
+            if not lines.empty():
+                content = view.substr(lines)
+                banner = self.make_banner(content, Prefs.banner_width, Prefs.character)
+                view.replace(edit, lines, banner)
+                view.sel().add(sublime.Region(lines.begin(), lines.begin() + len(banner)))
 
+        view.run_command('move', {'by': 'characters', 'forward': True})
+
+    def get_comment_charaters(self):

```

Comment:
You made a typo here, you probably intended to call it get_comment_chara__c__ters